### DISCUSSION TITLE
Language Recognition Issue

### DISCUSSION THREAD

**Discussion by Makuaa** [github-linguist/linguist#6494] [Classification] [answered]
Hello everyone,

I hope you're doing well. I'm currently facing an issue with language recognition for a specific file in my GitHub repository, and I thought it would be best to open a discussion to address it.

Repository: (https://github.com/Makuaa/Postman-API-Test)

I have added a .gitattributes file to specify the language for the Mypetstore.postman_collection.json file within my repository. The content of my .gitattributes file is as follows:

Mypetstore.postman_collection.json linguist-language=JSON   

Despite setting the language as JSON, the file is not being recognized correctly by Linguist, and it is not showing up as a JSON file in the repository.

I would like to request your assistance in resolving this issue. I believe that clearing the Linguist cache for my repository could potentially fix the problem. Could someone please guide me on how to proceed with clearing the cache or suggest any alternative solutions?

Thank you all for your time and support. I look forward to resolving this matter and improving the language recognition for my repository.

**Comment by lildude** [comment, marked as answer]
The file is being correctly identified as JSON (without the override) as you're getting correct syntax highlighting. The reason you're not seeing the language in the sidebar is because JSON is considered `data` so is excluded by default (see [How Linguist Works](https://github.com/github-linguist/linguist/blob/master/docs/how-linguist-works.md)).

If you want it to be considered in the stats and show in the sidebar you need to use the [detectable](https://github.com/github-linguist/linguist/blob/master/docs/overrides.md#detectable) override.

**Reply by Makuaa** [reply]
Thank you!

**Comment by Arantes19** [comment]
Hello dear git community,

Thanks in advance for your attention, I am also having problems with language recognition
Repository: https://github.com/Arantes19/EDA_2.1.2

Supposedly, I should have a much higher percentage of C language, but for some reason it's not being recognized, can you guys help me?
(I apologize if I'm reporting this in the wrong place)

Thank you all for your time and support.

**Reply by lildude** [reply]
@Arantes19 you should really start a new discussion for your issue.

The reason you're seeing so much HTML is because of all the files in the `Doxygen_Documentation/` directory. Linguist calculates the [percentage based on bytes of code](https://github.com/github-linguist/linguist/blob/master/docs/how-linguist-works.md) and all that HTML far outweighs your C. Of you don't want those  files to be considered, you should implement an [override](https://github.com/github-linguist/linguist/blob/master/docs/overrides.md).
